test/corev1alpha1_organization_test.go

```diff
@@ -74,45 +73,24 @@ func TestCoreOrganization(t *testing.T) {
 	require.NoError(t, cl.Create(ctx, org))
 	require.NoError(t, testutil.WaitUntilReady(ctx, cl, org))
 
-	projectTemplate := &corev1alpha1.OrganizationRoleTemplate{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{
-		Name: templates.ProjectAdminOrganizationRoleTemplateName,
-	}, projectTemplate))
-
-	rbacTemplate := &corev1alpha1.OrganizationRoleTemplate{}
-	require.NoError(t, cl.Get(ctx, types.NamespacedName{
-		Name: templates.RBACAdminOrganizationRoleTemplateName,
-	}, rbacTemplate))
-
-	assert.Empty(t, org.Status.Members)
-	rbacSubject := rbacv1.Subject{
-		Kind:     "User",
-		APIGroup: "rbac.authorization.k8s.io",
-		Name:     "User1",
-	}
-
-	rb := &rbacv1.RoleBinding{
+	projectTemplate := &corev1alpha1.OrganizationRoleTemplate{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "user1-rb",
-			Namespace: org.Status.Namespace.Name,
+			Name: templates.ProjectAdminOrganizationRoleTemplateName,
 		},
-		Subjects: []rbacv1.Subject{rbacSubject},
-		RoleRef: rbacv1.RoleRef{
-			APIGroup: "rbac.authorization.k8s.io",
-			Kind:     "Role",
-			Name:     "dummy",
+	}
+	require.NoError(t, testutil.WaitUntilReady(ctx, cl, projectTemplate))
+	rbacTemplate := &corev1alpha1.OrganizationRoleTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: templates.RBACAdminOrganizationRoleTemplateName,
 		},
 	}
-	require.NoError(t, cl.Create(ctx, rb))
+	require.NoError(t, testutil.WaitUntilReady(ctx, cl, rbacTemplate))
 	require.NoError(t, cl.WaitUntil(ctx, org, func() (done bool, err error) {
 		if len(org.Status.Members) > 0 {
-			assert.Equal(t, []rbacv1.Subject{rbacSubject}, org.Status.Members)
+			assert.Len(t, org.Status.Members, 1)
+			assert.Equal(t, org.Status.Members[0].Name, owner.Name)
 			return true, nil
 		}
 		return false, nil
 	}))
-	require.NoError(t, testutil.DeleteAndWaitUntilNotFound(ctx, cl, rb))
```

Comment:
This part of the test is import for testing dynamic reconciliation when rolebindings are added/deleted. Please don't just delete them, but adapt it.

Reply:
> This part of the test is import for testing dynamic reconciliation when rolebindings are added/deleted.

This is already tested by the currently e2e tests(since Owner is propagated to the `Status.Members`) isn't it?

Reply:
I have refactored this a bit to use the Owner impersonating client to create the `RoleBinding` so it can test that Owner really has permissions from the `OrganizationRoleTemplate`, PTAL